.github/workflows/create-tag.yml

```diff
@@ -0,0 +1,118 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'create-tag'
+on:
+  workflow_call:
+    inputs:
+      tag:
+        description: 'The name of the tag to be created.'
+        type: 'string'
+        required: true
+      annotated_tag:
+        description: 'Whether it will be an annotated tag in Git.'
+        type: 'boolean'
+        default: true
+        required: false
+      branch:
+        description: 'The branch to create tag from.'
+        type: 'string'
+        required: false
+        default: 'main'
+      message:
+        description: 'Message for the tag.'
+        type: 'string'
+        required: false
+      deployment_environment:
+        description: 'The environment with protection rules, see reference https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment'
+        type: 'string'
+        required: true
+      token_minter_wif_provider:
+        description: 'The full identifier of the Workload Identity Provider, including the project number, pool name, and provider name.'
+        type: 'string'
+        required: true
+      token_minter_wif_service_account:
+        description: 'Email address or unique identifier of the Google Cloud service account for which to generate credentials.'
+        type: 'string'
+        required: true
+      token_minter_service_audience:
+        description: 'Cloud Run audience for the production github-token-minter service'
+        type: 'string'
+        required: true
+      token_minter_service_url:
+        description: 'URL for the production github-token-minter service'
+        type: 'string'
+        required: true
+
+jobs:
+  create-tag:
+    runs-on: 'ubuntu-latest'
+    environment: '${{ inputs.deployment_environment }}'
+    steps:
+      - name: 'Mint Token'
+        id: 'mint-token'
+        uses: 'abcxyz/github-token-minter/.github/actions/mint-token@af661f898c6b0cda46dcaadd8299fb1eab264975' # ratchet:abcxyz/github-token-minter/.github/actions/mint-token@main
+        with:
+          wif_provider: '${{ inputs.token_minter_wif_provider }}'
+          wif_service_account: '${{ inputs.token_minter_wif_service_account }}'
+          service_audience: '${{ inputs.token_minter_service_audience }}'
+          service_url: '${{ inputs.token_minter_service_url }}'
+          requested_permissions: |-
+            {
+              "repositories": ["${{ github.event.repository.name }}"],
+              "permissions": {
+                "contents": "write"
+              }
+            }
+      - name: 'Create Tag'
+        uses: 'actions/github-script@98814c53be79b1d30f795b907e553d8679345975' # ratchet:actions/github-script@v6
+        with:
+          github-token: '${{ steps.mint-token.outputs.token }}'
+          retries: '3'
+          script: |+
+            let sha = '${{ github.sha }}'
+            if ('${{ inputs.branch }}' !== 'main') {
```

Comment:
I believe there is a script injection here. Dont trust anything from the users. Instead put this in an env var and use process.env.MY_VAR instead so it is escape. With the currently syntax anything can be rendered in there.

Reply:
Noted, thanks!